cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -321,3 +321,11 @@ func ManifestFilesAreEqual(path1, path2 string) (bool, error) {
 
 	return bytes.Equal(content1, content2), nil
 }
+
+// getProbeAddress returns a valid probe address
```

Comment:
this now needs a better explanation of why we are omitting the address here an why it works.

Reply:
to clarify, we need to explain why the k8s probes work with an empty address.
is this behavior guaranteed long term?

Reply:
@thockin is this something you can enlighten us? 
the question was if a probe will work with a 0.0.0.0  address https://github.com/kubernetes/kubernetes/pull/86493/files#r360576622
and we've found that (if my test is correct) it doesn't, however probes with empty hosts works 🤷‍♂

Reply:
alternatively, you can either a) check the source code of the probes, b) go the alternative route to not feed empty host - instead default to the link-local addresses.

Reply:
I've already done a) ;) https://github.com/kubernetes/kubernetes/pull/86505/
let's see how the discussion ends and based on the conclusion we can decide what's the best path for kubeadm

Reply:
@aojea i suggest we go with b) at this point.

Reply:
> to clarify, we need to explain why the k8s probes work with an empty address.
is this behavior guaranteed long term?

From the API doc: `// Host name to connect to, defaults to the pod IP.` so yes, it is guaranteed long term

Reply:
ok, let's proceed with merging this PR unless @aojea has reservations.
my minor request here is to document with at least another sentence above `getProbeAddress` why `""` works.

Reply:
Can't we go one step further and untangle the probe destination address from the bind address? We could always leave the probe to the empty host, so we could simplify this logic even more.

Reply:
if we are doing that it should be done for the kube-apiserver too.

Reply:
what if someone is expecting (with external tools) the probe address to be at the bind-address?

@aojea @ereslibre if you decide to go that route, please prefix the release note with "action required: ....".

Reply:
> what if someone is expecting (with external tools) the probe address to be at the bind-address?

Yeah, I'm fine with the PR as it is right now.

Forcing always the probe host to `""` (as I last proposed) has the drawback that if a user specifies the bind address host to a fixed IP address, then the probe wouldn't work (in this case, where we are using `hostNetwork`), so after a second thought I think it's better leaving the PR as is.